### PR TITLE
CI: Fix actions-310 testing 3.9 instead of 3.10

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed

### CODE REVIEW

**Comment by jbrockmendel** [comment]
test_write_ignoring_index?

**Comment by jbrockmendel** [comment]
LGTM cc @jreback should get us back to green?

**Comment by mroeschke** [comment]
> should get us back to green?

Not necessarily. Just an error in our CI configs.

File: pandas/tests/io/test_fsspec.py [hunk 1]
```diff
@@ -181,6 +182,7 @@ def test_arrowparquet_options(fsspectest):
 
 @td.skip_array_manager_not_yet_implemented  # TODO(ArrayManager) fastparquet
 @td.skip_if_no("fastparquet")
+@pytest.mark.xfail(PY310, reason="fastparquet failing on 3.10")
```
Comment (jreback) [inline pandas/tests/io/test_fsspec.py:185]: do we have an issue for this?

File: pandas/tests/io/test_parquet.py [hunk 2]
```diff
@@ -431,27 +440,39 @@ def test_columns_dtypes_invalid(self, engine):
...
+            request.node.add_marker(
+                pytest.mark.xfail(reason="fastparquet failing on 3.10")
+            )
         df = pd.DataFrame({"A": [1, 2, 3]})
         check_round_trip(df, engine, write_kwargs={"compression": compression})
 
-    def test_read_columns(self, engine):
+    def test_read_columns(self, engine, request):
         # GH18154
+        if PY310 and engine == "fastparquet":
```
Comment (jreback) [inline pandas/tests/io/test_parquet.py:460]: can you just do this in the engine fixture itself?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/test_fsspec.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/io/test_fsspec.py:185]: Created https://github.com/pandas-dev/pandas/issues/45605

File: pandas/tests/io/test_parquet.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/io/test_parquet.py:460]: Appears some of the tests parameterize with `engine="fastparquet"` pass and some fail, so these need to be targeted.

**Comment by mroeschke** [comment]
Azure failures are the flaky timeouts

File: pandas/tests/io/test_fsspec.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/io/test_fsspec.py:185]: can you add this issue number to the msgs :->

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@meeseeksdev backport 1.4.x

**Pull request merged**